### DISCUSSION TITLE
Making multiple custom variants chainable

### DISCUSSION THREAD

**Discussion by tandrewnichols** [tailwindlabs/tailwindcss#3559] [Help]
I'm using [tailwindcss-important](https://github.com/chasegiunta/tailwindcss-important) to generate an important variant (tl;dr it's basically identical to the example in the tailwind docs for generating an important variant). Separately, I wrote a plugin to generate before and after pseudo selectors using `after:classname` and `before:classname`. Both plugins work fine on their own. However, in some cases, when trying to use before/after, some default styles take precedence. For example, `after:top-4` but some other class on the same element has specified top already, so my style isn't applied. What I would like to be able to do is add `!after:top-4` as a class (to add `!important` to the `after` classes essentially). I thought just ordering the variants correctly in `theme.inset` would be sufficient, but it doesn't work (I've tried both orders, `important` and then `after` as well as `after` followed by `important`). So now I'm assuming it must be something about the way these specific selectors are written that makes this not work. But I'm at a loss to know where to start in _making_ it work. Is there a simple way to make the two work together? For reference, here are the two plugins as they stand now:

**important**

```
module.exports = function() {
  return function({ addVariant }) {
    addVariant('important', ({ container }) => {
      container.walkRules(rule => {
        rule.selector = `.\\!${rule.selector.slice(1)}`
        rule.walkDecls(decl => {
          decl.important = true
        })
      })
    })
  }
}
```

I just noticed that this doesn't use `plugin` from tailwindcss, but it _seems_ to work other than in the case I'm talking about here.

**after**

```
const plugin = require('tailwindcss/plugin');

module.exports = () => {
  return plugin(({ addVariant, e }) => {
    addVariant('after', ({ modifySelectors, separator }) => {
      modifySelectors(({ className }) => {
        return `.${e(`after${separator}${className}`)}:after`;
      });
    });
  });
};

```

**Comment by pintend** [comment]
```after:!top-4```

**Comment by tandrewnichols** [comment]
Ah, interesting. I hadn't thought of that. I'm on a completely different team now that is primarily backend work, so I don't know how soon I'd be able to try this, but I'm assuming you either know for a fact this works or have tried it to confirm. Thanks!
